source/extensions/filters/network/dubbo_proxy/app_exception.h

```diff
@@ -2,30 +2,48 @@
 
 #include "envoy/common/exception.h"
 
-#include "extensions/filters/network/dubbo_proxy/deserializer.h"
 #include "extensions/filters/network/dubbo_proxy/filters/filter.h"
 #include "extensions/filters/network/dubbo_proxy/metadata.h"
 #include "extensions/filters/network/dubbo_proxy/protocol.h"
+#include "extensions/filters/network/dubbo_proxy/serializer.h"
 
 namespace Envoy {
 namespace Extensions {
 namespace NetworkFilters {
 namespace DubboProxy {
 
-struct AppException : public EnvoyException,
-                      public DubboFilters::DirectResponse,
-                      Logger::Loggable<Logger::Id::dubbo> {
-  AppException(ResponseStatus status, const std::string& what);
-  AppException(const AppException& ex) = default;
+using ResponseType = DubboFilters::DirectResponse::ResponseType;
 
-  using ResponseType = DubboFilters::DirectResponse::ResponseType;
-  ResponseType encode(MessageMetadata& metadata, Protocol& protocol, Deserializer& deserializer,
-                      Buffer::Instance& buffer) const override;
+template <typename T = ResponseStatus>
+struct AppExceptionBase : public EnvoyException,
+                          public DubboFilters::DirectResponse,
+                          Logger::Loggable<Logger::Id::dubbo> {
+  AppExceptionBase(const AppExceptionBase& ex) = default;
+  AppExceptionBase(T status, const std::string& what)
+      : EnvoyException(what), status_(status),
+        response_type_(RpcResponseType::ResponseWithException) {}
 
-  const ResponseStatus status_;
+  ResponseType encode(MessageMetadata& metadata, DubboProxy::Protocol& protocol,
+                      Buffer::Instance& buffer) const override {
+    ASSERT(buffer.length() == 0);
```

Comment:
I can't append data to a buffer?

Reply:
yes, at present, I haven't thought of any scenarios that require append data, and I can let go of this restriction if necessary.